apps/server/src/modules/tool/service/external-tool.service.ts

```diff
@@ -44,6 +39,29 @@ export class ExternalToolService {
 		return created;
 	}
 
+	async updateExternalTool(toUpdate: ExternalToolDO): Promise<ExternalToolDO> {
+		if (toUpdate.config instanceof Oauth2ToolConfigDO) {
+			const toUpdateOauthClient: ProviderOauthClient = this.mapper.mapDoToProviderOauthClient(
```

Comment:
reading this, im wondering if this is actually a transformation of the external tool into a oauthprovider, or rather the construction of a client fitting to the tool. In other words, if that mapper should be some sort of factory instead. But thats just a thought, im not even sure myself.

Reply:
To be honest I'm not sure either. I'll think about it